core/src/main/java/com/xatkit/core/recognition/nlpjs/NlpjsService.java

```diff
@@ -0,0 +1,113 @@
+package com.xatkit.core.recognition.nlpjs;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.xatkit.core.recognition.IntentRecognitionProviderException;
+import com.xatkit.core.recognition.nlpjs.adapter.ResolutionDeserializer;
+import com.xatkit.core.recognition.nlpjs.model.*;
+import fr.inria.atlanmod.commons.log.Log;
+import lombok.NonNull;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.concurrent.TimeUnit;
+
+public class NlpjsService {
+
+    private final static String NLPJS_BASE_PATH = "/api";
+
+    private String nlpjsServer;
+
+    private NlpjsApi nlpjsApi;
+
+    private Gson gson;
+
+    public NlpjsService(@NonNull String nlpjsServer) {
+        this.nlpjsServer = nlpjsServer;
+        gson = new GsonBuilder().registerTypeAdapter(Resolution.class, new ResolutionDeserializer())
+                .create();
+        String nlpjsApiFullPath = this.nlpjsServer + NLPJS_BASE_PATH + "/";
+        OkHttpClient httpClient = new OkHttpClient.Builder()
+                .addInterceptor(new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
+                .readTimeout(3, TimeUnit.SECONDS)
+                .build();
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(nlpjsApiFullPath)
+                .client(httpClient)
+                .addConverterFactory(GsonConverterFactory.create(gson)).build();
+        this.nlpjsApi = retrofit.create(NlpjsApi.class);
+    }
+
+
+    public Agent getAgentInfo(String agentId) throws IOException, IntentRecognitionProviderException {
+        Response<Agent> agentResponse = nlpjsApi.getAgentInfo(agentId).execute();
+        if (!agentResponse.isSuccessful()) {
+            ErrorBody errorBody = gson.fromJson(agentResponse.errorBody().string(), ErrorBody.class);
+            String errorMessage = MessageFormat.format("Unsuccessful REST operation {0} " +
+                            "on {1}. The API responded with the status {2} and the error message \"{3}\" ",
+                    nlpjsApi.getAgentInfo(agentId).request().method(),
+                    nlpjsApi.getAgentInfo(agentId).request().url(), agentResponse.code(), errorBody.getMessage());
+            Log.error(errorMessage);
+            throw new IntentRecognitionProviderException(errorMessage);
+        }
+        return agentResponse.body();
+    }
+
+
+    public boolean createAgent(AgentInit agentInit) throws IOException, IntentRecognitionProviderException {
```

Comment:
This method is never used. I know you added support for multiple agent so I'd keep it around even if it's not used at the moment, but we need a piece of documentation to explain that.

Reply:
Indeed, it's never used. We may need it in the future (e.g., create an agent if it does not exit)